### PR TITLE
feat(nats): add flag to indicate that nats is not connected

### PR DESCRIPTION
**Pull request by dwojno**

### CODE REVIEW

**Review by kim-energyweb — APPROVED** [review]
LGTM

File: src/iam/iam-base.ts [hunk 1]
```diff
@@ -337,12 +337,25 @@ export class IAMBase {
...
-      let protocol = "ws";
-      if (this._natsServerUrl.indexOf("https://") === 0) {
-        protocol = "wss";
+      try {
+        let protocol = "ws";
+        if (this._natsServerUrl.indexOf("https://") === 0) {
+          protocol = "wss";
+        }
+
+        const connection = await Promise.race<NatsConnection | undefined>([
```
Comment (jrhender) [inline src/iam/iam-base.ts:346]: It looks to me like this is adding kind-of a back-up timeout to the connect. Is that right? As it looks like the `connect` constructor already has a timeout. Why the need for the duplication? Is it that the `connect` timeout is unreliable?
Let me know if I'm missing something or it's a common pattern that I just don't know 😅

Comment (dwojno) [reply src/iam/iam-base.ts:346]: xD yeah, this NATS.ws is throwing error on timeout that I'm not able to catch ;) I've tried all things I can think off, try/catch in all places, .catch(), nothing works, so I've added a race condition timeout to make it work :)

File: src/iam/iam-base.ts [hunk 2]
```diff
@@ -337,12 +337,25 @@ export class IAMBase {
...
+          connect({
+            servers: `${protocol}://${this._natsServerUrl}`,
+            timeout: 2000
```
Comment (jrhender) [inline src/iam/iam-base.ts:349]: Minor point, but I would consider assigning `2000` to a variable so that both timeouts can be sure to use the same thing

**Review by jrhender — APPROVED** [review]
Looks good

File: src/iam/iam-base.ts (hunk 1 above)
Comment (dwojno) [reply src/iam/iam-base.ts:347]: 👍

**Pull request merged**